### PR TITLE
feat: worker pool based event routing

### PR DESCRIPTION
**Pull request by jyecusch**

### CODE REVIEW

File: pkg/cmd/run/root.go [hunk 1]
```diff
@@ -85,6 +85,12 @@ var runCmd = &cobra.Command{
 		gw, err := run.NewGateway()
 		cobra.CheckErr(err)
 
+		ev, err := run.NewEvents(pool)
+
+		if err != nil {
```
Comment (davemooreuws) [inline pkg/cmd/run/root.go]: doesnt require if statement

**Review by davemooreuws — CHANGES_REQUESTED** [review]

**Review by tjholm — APPROVED** [review]
looks good to me apart from Dave's requested change and a copyright header that needs fixing.

File: pkg/provider/run/events.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+			"event": event,
+		},
+	)
+
+	requestId := event.ID
+	// payloadType := event.PayloadType
+	payload := event.Payload
+
+	marshaledPayload, err := json.Marshal(payload)
+
```
Comment (asalkeld) [inline pkg/provider/run/events.go]: can we not leave a space between the point of error creation and it's check?
This just makes it clearer what the check is associated with, and doesn't make the code quite so spread out.
like so :
```
x, err = foo()
if err != nil {
    return err
}
```

File: pkg/provider/run/events.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	targets := s.pool.GetWorkers(&worker.GetWorkerOptions{
+		Event: evt,
+	})
+
+	fmt.Println(fmt.Sprintf("Publishing event to: %s", targets))
+	for _, target := range targets {
+		err = target.HandleEvent(evt)
+
+		if err != nil {
+			fmt.Println(err)
```
Comment (asalkeld) [inline pkg/provider/run/events.go]: so this is just a "user error"? - so not worth changing the flow of our program..

**Review by asalkeld — COMMENTED** [review]
just stylistic nits

File: pkg/cmd/run/root.go (hunk 1 above)
Comment (jyecusch) [reply pkg/cmd/run/root.go]: fixed

File: pkg/provider/run/events.go (hunk 3 above)
Comment (jyecusch) [reply pkg/provider/run/events.go]: fixed, added a comment so that intention isn't lost to the ages :)

**Review by davemooreuws — APPROVED** [review]
Looks good just fix the lint

**Pull request merged**